WordPress/gutenberg
# ScrollLock: Covert component to TypeScript

## What?
Converts the `ScrollLock` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `ScrollLock` to TypeScript.
- Refactors the test to `@testing-library/react` and TypeScript
- Refactors storybook to TypeScript.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `ScrollLock` continues to function as expected
